.github/workflows/changed-files.yaml

```diff
@@ -0,0 +1,78 @@
+on:
+  workflow_call:
+    inputs:
+      files_yaml:
+        type: string
+        required: true
+      transform_expr:
+        type: string
+        required: false
+        default: |
+          to_entries |
+          map(
+            select(.key | endswith("_any_changed")) |
+            {
+              "key": (.key | rtrimstr("_any_changed")),
+              "value": (.value == "true"),
+            }
+          ) |
+          from_entries
+    outputs:
+      changed_file_groups:
+        value: ${{ jobs.changed-files.outputs.transformed_output }}
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
```

Comment:
Review this. Do we need “read” on packages?

Reply:
Probably not. I'm also questioning whether we need `id-token: write`.

Requesting input from @ajschmidt8.

Reply:
We can revisit this later. I'll go ahead and merge.

Reply:
> Probably not. I'm also questioning whether we need `id-token: write`.
> 
> Requesting input from @ajschmidt8.

No, `id-token: write` is not necessary here. It's only needed in workflows that require AWS credentials via [aws-actions/configure-aws-credentials](https://github.com/aws-actions/configure-aws-credentials/).

Can this branch be deleted now that the PR is merged?

Reply:
I went ahead and deleted it.